### PR TITLE
Fix typos in the code base

### PR DESCRIPTION
**Pull request by jbrianceau**
## Description
Fix few typos in the code base.
Note that only documentation and comments are impacted by this change.

### CODE REVIEW

File: extras/catch_amalgamated.cpp [hunk 1]
```diff
@@ -3550,7 +3550,7 @@ namespace Catch {
             // Extracts the actual name part of an enum instance
             // In other words, it returns the Blue part of Bikeshed::Colour::Blue
             StringRef extractInstanceName(StringRef enumInstance) {
-                // Find last occurence of ":"
```
Comment (horenmar) [inline extras/catch_amalgamated.cpp]: Neither of the `extras/catch_amalgamated.{h,c}pp` files should be touched outside of creating new release.

**Comment by horenmar** [comment]
Thanks, but if you want the changes to propagate to cppwinrt, you should probably run a similar PR against the `v2.x` branch.

File: extras/catch_amalgamated.cpp (hunk 1 above)
Comment (jbrianceau) [reply extras/catch_amalgamated.cpp]: Oh I missed that, I've just updated the PR to revert the change on this file.

**Comment by jbrianceau** [comment]
> Thanks, but if you want the changes to propagate to cppwinrt, you should probably run a similar PR against the `v2.x` branch.

This pull request is about quite cosmetic changes, so I don't think it's worth doing a backport of it for the `v2.x` branch.
Still, let me know if you think this backport would be nice to have, and I'll create a PR against this branch in that case.

**Comment by horenmar** [comment]
I think that having a separate PR for v2 would be useful.

**Comment by jbrianceau** [comment]
> I think that having a separate PR for v2 would be useful.

Sure, here it is #2208

**Pull request merged**